Data_Structures/Linked_List/Floyd_Cylce_detection/floyd_cycle_detection.swift

```diff
@@ -0,0 +1,82 @@
+//  floyd_cycle_detection.swift
+//
+//  Swift implementation of a linked list with cycle detection functionality
+//
+//  How to run: type on terminal `swift floyd_cycle_detection.swift` and type enter. 
+//              The test program will populate two lists, one with loop and one without
+//              and will print test both for if theres a loop or not.
+
+// A class representing one node of the list
+class Node<T> {
+
+    // public properties
+    public var value: T
+    public var next: Node<T>?
+
+    // Create a new node with a given value
+    public init (value: T) {
+        self.value = value
+    }
+
+    // Insert an element at the end of the list recursively
+    public func insert(_ node: Node<T>) {
```

Comment:
It is an elegant way to add, but if the list became too large, maybe this log(N) insertion will be expensive. What you thing about a List structure with Head and Tail com encapsulate this cost?

Reply:
Also, what if submit the subject of the list ("T") instead Note.